main.cpp

```diff
@@ -108,7 +107,19 @@ Thing 4) bass guitar amp
     3) adjust tweeter speaker mix
  */
 
-
+/*
+Thing 10) compressor
+5 properties:
+    1) ratio
+    2) attack time
+    3) release time
+    4) makeup gain
+    5) threshold
+3 things it can do:
+    1) control dynamic range
+    2) makeup low volume signal
+    3) increase sustain
```

Comment:
The 10th UDT needs to have 5 properties that are themselves Objects.   You currently have 5 properties that can be represented with Primitives. attackTime, releaseTime, makeupGain can all be represented with `float` or `double`.
Take another look at instruction 1)
- [ ] revise